extmod/modssl_mbedtls.c

```diff
@@ -276,6 +454,41 @@ STATIC void ssl_context_load_cadata(mp_obj_ssl_context_t *self, mp_obj_t cadata_
     mbedtls_ssl_conf_ca_chain(&self->conf, &self->cacert, NULL);
 }
 
+#ifdef MICROPY_SSL_MBEDTLS_EXTRAS
+STATIC mp_obj_t mod_ssl_load_cadata(size_t n_args, const mp_obj_t *pos_args,
+    mp_map_t *kw_args) {
+
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_cafile, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_rom_obj = MP_ROM_NONE} },
+        { MP_QSTR_cadata, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_rom_obj = MP_ROM_NONE} },
+    };
+
+    mp_obj_ssl_context_t *self = MP_OBJ_TO_PTR(pos_args[0]);
+    mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
+    mp_arg_parse_all(n_args - 1, pos_args + 1, kw_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
+
+    // cafile
+    if (args[0].u_obj != mp_const_none) {
+        mp_obj_t cadata;
+
+        if (!(mp_obj_is_type(args[0].u_obj, &mp_type_bytes))) {
+
+            cadata = read_file(args[0].u_obj);
+            ssl_context_load_cadata(self, cadata);
+        }
+
+    }
+    // cadata
+    else if (args[1].u_obj != mp_const_none) {
```

Comment:
I think CPython allows specifying both cafile and cadata at the same time, so this can just be `if` (not `else if`).